src/main/java/mate/academy/springboot/datajpa/controller/CategoryController.java

```diff
@@ -0,0 +1,57 @@
+package mate.academy.springboot.datajpa.controller;
+
+import mate.academy.springboot.datajpa.dto.CategoryRequestDto;
+import mate.academy.springboot.datajpa.dto.CategoryResponseDto;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.mapper.RequestDtoMapper;
+import mate.academy.springboot.datajpa.service.mapper.ResponseDtoMapper;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final ResponseDtoMapper<CategoryResponseDto, Category> categoryResponseMapper;
+    private final RequestDtoMapper<CategoryRequestDto, Category> categoryRequestMapper;
+
+    public CategoryController(CategoryService categoryService,
+                              ResponseDtoMapper<CategoryResponseDto,
+                                      Category> categoryResponseMapper,
+                              RequestDtoMapper<CategoryRequestDto,
+            Category> categoryRequestMapper) {
+        this.categoryService = categoryService;
+        this.categoryResponseMapper = categoryResponseMapper;
+        this.categoryRequestMapper = categoryRequestMapper;
+    }
```

Comment:
You can use annotation from lombok that creates the same constructor.
See there are several variants